examples/hpc/dla-driver-ffi/src/lib.rs

```diff
@@ -459,28 +436,20 @@ pub unsafe extern "C" fn dla_tvm_qnn_conv2d(
         None,
     );
 
+    let input_order_string = unsafe { CStr::from_ptr(input_order).to_str().unwrap_unchecked() };
+
     // TVM requantization and clip
     // NOTE:(20240927 vaino-waltteri.granat@tuni.fi) on DLA clipping behaviour with TVM.
     // DLA's conv2d arithmetic is done at 16 bit width, but the output of the DLA is limited to 8 bits.
     // To comply with TVM's expected value range our solution is to bit shift/clip the 16-bit result of
     // conv2d by 8 bits and shift if back in the driver. This causes some amount of data loss due to
     // the lost granularity of the values. The clipping amount is set by the pp_clip argument.
-    rescale(
-        &mut result,
-        u32::pow(2, pp_clip) as f32,
-        input_zero[0],
-        output_zero[0],
-        input_scale[0],
-        output_scale,
-    );
-
-    let input_order_string = unsafe { CStr::from_ptr(input_order).to_str().unwrap_unchecked() };
+    let mut res_i32: Vec<i32> = result.to_buffer_with_order(Order3::try_from(input_order_string).unwrap_unchecked())
+                                       .iter().map(|x: &i8| (*x as f32 * u32::pow(2, pp_clip) as f32) as i32).collect();
```

Comment:
By the way, if this is related to the hardware bug, one thing you could do is to `#[cfg(feature = "asic")]`. That way we can still get accurate results from the VP while addressing the issues with the ASIC.

Reply:
OTOH it can be good to be as close to hardware implementation as possible with the VP as well, bugs and all included. idk, food for thought :)

Reply:
It's not really a hardware bug, but rather the way hardware is designed. DLA-VP implements this hardware behavior to ensure compatibility with RTL tests. More on this in my thesis :D